### DISCUSSION TITLE
Can I use EdgeConv for bipartite graphs with different feature dimensions?

### DISCUSSION THREAD

**Discussion by kylemin** [pyg-team/pytorch_geometric#6437] [Q&A] [answered]
The GNN Cheatsheet (https://pytorch-geometric.readthedocs.io/en/latest/notes/cheatsheet.html) says that EdgeConv supports message passing in bipartite graphs with potentially different feature dimensionalities. Can someone help me figure this out?

For example, let's say we have a heterogenous graph where there are two types of nodes ({x} and {y}) and the x's feature dimension is 64 and the y's feature dimension is 32. In this case, we can apply HeteroConv as below:

hetero_conv = HeteroConv({
    ('x', 'xx', 'x'): SAGEConv((64, 64), 32),
    ('y', 'yy', 'y'): SAGEConv((32, 32), 32),
    ('x', 'xy', 'y'): SAGEConv((64, 32), 32),
}, aggr='sum')

We can also change the first two SAGEConv layers to EdgeConv. However, for ('x', 'xy', 'y'), we cannot use EdgeConv because x and y have different dimensionalities. Is this correct? Or is there any other way that I can use EdgeConv for this situation?

**Comment by wsad1** [comment, marked as answer]
It looks like you can't use [EdgeConv](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.conv.EdgeConv). It supports bipartite graphs but the feature dimensions have to be the same.

**Reply by kylemin** [reply]
Thank you for the reply!

**Comment by rusty1s** [comment]
Yes, in this case you would first map them to the same space via separate `Linear` modules.

**Reply by kylemin** [reply]
I see, thank you for the suggestion!
